R/zchunk_LB1322.Fert.R

```diff
@@ -45,88 +57,386 @@ module_energy_LB1322.Fert_DISABLED <- function(command, ...) {
+      mutate(in_Fert = Fert_Prod_MtN * intensity_GJkgN) %>%
+      select(iso, year, fuel, value, value_share, in_Fert, intensity_GJkgN) ->
+      L1322.Fert_Prod_MtN_ctry_F_Y
+
+    # Checking total energy inputs to fertilizer against industrial energy use for each region
+    # First, create a couple lists to use to expand table in a later step
+    list_GCAM_region_ID <- unique(iso_GCAM_regID$GCAM_region_ID)
+    list_fuels <- unique(L1322.Fert_Prod_MtN_ctry_F_Y$fuel)
+
+    # Aggregate by GCAM region
+    L1322.Fert_Prod_MtN_ctry_F_Y %>%
+      left_join_error_no_match(iso_GCAM_regID, by = "iso") %>% # Assign data to GCAM region using iso
+      group_by(GCAM_region_ID, fuel, year) %>% # Aggregate by GCAM region, fuel, and year
+      summarise(in_Fert = sum(in_Fert)) %>%
+      ungroup() %>%
+      # Expand this table to all regions, fuels, and years in case there are regions that aren't countries in the aglu databases
+      complete(GCAM_region_ID = list_GCAM_region_ID,
+               fuel = list_fuels,
+               year = HISTORICAL_YEARS,
+               fill = list(in_Fert = 0)) %>%
+      # Attach data from ind energy and ind feedstocks
+      left_join_error_no_match(L1321.in_EJ_R_indenergy_F_Yh, by = c("GCAM_region_ID", "year", "fuel")) %>%
+      left_join_error_no_match(L132.in_EJ_R_indfeed_F_Yh, by = c("GCAM_region_ID", "fuel", "year")) %>%
+      select(GCAM_region_ID, fuel, year, in_Fert, in_indenergy = value.x, in_indfeed = value.y) %>%
+      # Finally, check whether the remaining available industrial energy use is not negative
+           # (i.e., ind. energy + ind. feedstocks - fert)
+      mutate(check = in_indenergy + in_indfeed - in_Fert) ->
+      L1322.Fert_ALL_MtN_R_F_Y
+
+    # Modifying fertilizer production shares so that industrial energy/feedstock use does not go negative for any fuels
+    # Modify the fuel shares in the country-level fertilizer production tables, and re-compute fertilizer production quantities
+    L1322.Fert_ALL_MtN_R_F_Y %>%
+      filter(check < 0) %>%
+      mutate(mult = (1 - (-1 * check / in_Fert))) %>%
+      select(GCAM_region_ID, fuel, year, mult) ->
+      L1322.Fert_modify
+
+    # For matched region/fuel/year, multiply the original shares by the multipliers from above
+    L1322.Fert_Prod_MtN_ctry_F_Y %>%
+      left_join_error_no_match(iso_GCAM_regID, by = "iso") %>%
+      # left_join_error_no_match cannot be used because there are some rows in the base table left unmatched. NAs will be introduced.
+      left_join(L1322.Fert_modify, by = c("GCAM_region_ID", "fuel", "year")) %>%
+      replace_na(list(mult = 1)) %>%
+      mutate(value_share_adj = value_share * mult) %>%
+      select(iso, GCAM_region_ID, year, fuel, value, value_share_adj, intensity_GJkgN) ->
+      L1322.Fert_Prod_MtN_ctry_F_Y_share_adj
+
+    # For regions whose fuel shares changed, re-assign the production and energy inputs to the oil-based technology
```

Comment:
I checked, and the sum of coal and gas shares never exceed 1. So the minimum share for refined liquids would be 0.

I think there are checks before that, but I could probably include something here. If the sum of coal and gas shares exceed 1, then I would have to change not only refined liquids but also coal and gas.